src/lib/snarky_group_hash/snarky_unchecked_hash.ml

```diff
@@ -0,0 +1,342 @@
+(* Points on elliptic curves over finite fields by M. SKALBA
+ * found at eg https://www.impan.pl/pl/wydawnictwa/czasopisma-i-serie-wydawnicze/acta-arithmetica/all/117/3/82159/points-on-elliptic-curves-over-finite-fields
+ *
+ * Thm 1.
+ * have f(X1)f(X2)f(X3) = U^2 => one on f(X1), f(X2) or f(X3) is square.
+ * take that Xi as the x coordinate and solve for y ( which can be done
+ * because f(Xi) is square :) )
+ *
+ * Thm 2.
+ * if we take map(t) = (Xj(t^2), sqrt(f(Xj(t^2)),
+ * with j = min{1 <= i <= 3 | f(Xi(t^2)) in F_q^2}, then map(t)
+ * is well defined for at least |T| - 25 values of t and |Im(map| > (|T|-25)/26
+ *)
+
+open Core
+
+module Make_snarky_unchecked_hash
+    (F : Snarky.Field_intf.Extended) (Params : sig
+        val a : F.t
+
+        val b : F.t
+    end) =
+struct
+  module Hash_key = struct
+    module T = struct
+      type t = int * int [@@deriving compare, hash, sexp]
+
+      let equal = ( = )
+    end
+
+    include T
+    include Comparable.Make (T)
+    include Hashable.Make (T)
+  end
+
+  let n1_tuples =
+    [ ((0, 0), "212")
+    ; ((0, 1), "-208")
+    ; ((3, 0), "-161568")
+    ; ((0, 2), "-264")
+    ; ((3, 1), "441408")
+    ; ((0, 3), "304")
+    ; ((6, 0), "-92765376")
+    ; ((3, 2), "-127776")
+    ; ((0, 4), "-44") ]
```

Comment:
You could avoid having the list and hash table defs separate by doing e.g.,

`Hash_key.Table.of_alist_exn [ ... ]`